projects/cashmere/src/lib/app-switcher/app-switcher.service.spec.ts

```diff
@@ -0,0 +1,150 @@
+import {TestBed, async, inject} from '@angular/core/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {AppSwitcherService} from './app-switcher.service';
+import {APP_SWITCHER_CONFIG} from './app-switcher-interfaces';
+
+describe('AppSwitcherService', () => {
+    describe('when initialized with a discoveryServiceUri ending in a version', () => {
+        let service: AppSwitcherService;
+
+        beforeEach(async(() => {
+            TestBed.configureTestingModule({
+                imports: [HttpClientTestingModule],
+                providers: [AppSwitcherService, {provide: APP_SWITCHER_CONFIG, useValue: {discoveryServiceUri: 'foo://bar/baz/v1'}}]
+            });
+        }));
+
+        beforeEach(() => (service = TestBed.get(AppSwitcherService)));
+
+        it('should have the correct allApplicationsUri', () => {
+            expect(service.allApplicationsUri).toBe('foo://bar/baz/apps');
+        });
+
+        it('should use the correct uri when fetching services', async(
+            inject([HttpTestingController], (httpMock: HttpTestingController) => {
+                service.getApplications().subscribe();
+
+                const req = httpMock.expectOne(
+                    `foo://bar/baz/v1/Services?$filter=DiscoveryType eq 'Application' and IsHidden eq false&$top=12`,
```

Comment:
I know this library is mostly for our own internal use, but seems like we could configure this in a better way to give users better control over how these are fetched instead of assuming things about the service.

Nothing to change, just curious what everyone thinks

Reply:
Currently there is only one valid way to use the DiscoveryService; I imagine that in the future that could change (with v2, etc).  I would like to see IAppSwitcherConfig to include more options.  Something possibly like this: (This would be great to include in a future major version):

```ts
export interface IAppSwitcherConfig {
    discoveryServiceUri: string;
    paths: IAppSwitcherPaths;
}

export interface IAppSwitcherPaths {
    allAppsPath: string;
    getApplicationsPath: string
}

export const VERSION_1_PATHS: IAppSwitcherPaths = {
    allAppsPath: '/apps',
    getApplicationsPath: '/v1/Services?...'
};
```

I would also like to rename AppSwitcherService to DiscoveryAppSwitcherService since it is coupled to the Discovery Service.

@jonest and @benjanderson what do you think (for future versions of course)?

Reply:
That looks like a really good next iteration to me!

Depending on how open we want this OSS project to be, we could just provide a callback to get the apps list and pull out the entire idea of a discovery service where the shape of the app model is unique to our service.